src/main/java/parsing/Parser.java

```diff
@@ -63,18 +71,17 @@ public NodeGraph parse(File file) {
 
     /**
      * Parses a .gfa file to a graph.
-     * @param file The name of the target .gfa file.
+     * @param file  The name of the target .gfa file.
      * @param graph The graph the data gets put into.
      * @return The graph created from the .gfa file.
      */
     public NodeGraph parse(final File file, NodeGraph graph) {
```

Comment:
This method is rather complex, is there any way you can split it into multiple smaller methods?